JavaScript/JavaScriptSDK.Tests/CheckinTests/Context/Session.tests.ts

```diff
@@ -376,5 +466,67 @@ class SessionContextTests extends TestClass {
     private restoreFakeCookie() {
         Microsoft.ApplicationInsights.Util["document"] = this.originalDocument;
     }
+
+    // Make a fake document.cookie property that correctly handles its unique getter/setter functionality.
+    // Document.cookie lets you read the entire set of cookies, but on set acts essentially as +=, with
+    // some logic to overwrite existing cookies with the same name.
+    // Document.cookie also silently removes expired cookies.
+    private generateEmulatedCookieStore() {
+        var cookieStore = {
+            _cookies: <string[]>[]
+        };
+        Object.defineProperty(cookieStore, "cookie", {
```

Comment:
It's possible to stub the functions that actually write to the cookie, though it seemed like it would leave the greatest amount of the system unchanged, and by my thinking more fully tested, by emulating the browser api and not impacting any of the internal functions directly. The current way we test cookies in the other tests will generate confusing discrepancies in all but the most simple cases as we replace document.cookie with a simple string. I think we should consider actually replacing the document.cookie replacements we do in other places with this to avoid other cases where our tests will behave differently than real browsers.